bears/c_languages/CPPCheckBear.py

```diff
@@ -28,18 +29,21 @@ class CPPCheckBear:
     CAN_DETECT = {'Security', 'Unused Code', 'Unreachable Code', 'Smell'}
 
     @staticmethod
-    def create_arguments(filename, file, config_file,
+    def create_arguments(config_file,
+                         path_to_dir: str,
                          enable: typed_list(str)=[]):
         """
         :param enable:
             Choose specific issues to report. Issues that can be
             reported are: all, warning, style, performance,
             portability, information, unusedFunction,
             missingInclude
+        :param path_to_dir:
+            Path to the directory where analysis to be done.
```

Comment:
nope, please look at the signatures of other global-bear-linters, coala feeds in all files in a file-dict. Having a setting for this is definitely bad design, then you have to do
```
# in a coafile
files = **
path_to_dir = this_dir  # <-- hmm that's what actually "files" is for
```

Reply:
I have tried this
`files = this_dir`
then coala says
No files matching `this_dir` were found.

Reply:
@Makman2 
Because --files parameter need direct path to files not to directory.May be it can be done using glob pattern like this
`files = this_dir/**.cpp, this_dir/**.c`
Is it right to do like this?

Reply:
**.c will assure that all the c files in any depth will be taken care of so, IMO this_dir/**.c didn't make much sense, still lets see what makman says

Reply:
@newbazz this linter analyze C and CPP files.So If this way that I propose is right then I think there is no problem in 
`files = this_dir/**.cpp, this_dir/**.c`

Reply:
that's the way we go, the user should specify all C/C++ files and you pass all of them to the linter.

Reply:
@Makman2 But the problem is how I can get the value of --files argument in `create_argument` method because while using linter for writing globalbear then only allowed option are config_file, and extra setting that we add.

Reply:
@NiklasMM ^ any help, please

Reply:
Well, by design a global bear does not need to be told which files it should look at, because it looks at the whole repo. That's what defines a global bear. You should be fine with just using the current directory `.`. Otherwise using a global bear is not the correct solution.